### PR TITLE
Add Analytics

### PR DESCRIPTION
**Pull request by nemesv**
Implements #43

Docs: https://github.com/SpecFlowOSS/SpecFlow.Rider/blob/analytics/doc/how-it-works.md#analytics

### CODE REVIEW

File: src/dotnet/ReSharperPlugin.SpecflowRiderPlugin/Analytics/AppInsightsEventTelemetry.cs [hunk 1]
```diff
@@ -0,0 +1,100 @@
+using System;
+using System.Collections.Generic;
+using System.Runtime.InteropServices;
+using Newtonsoft.Json;
+
+namespace ReSharperPlugin.SpecflowRiderPlugin.Analytics
+{
+    /// <summary>
```
Comment (alexandrafung) [inline src/dotnet/ReSharperPlugin.SpecflowRiderPlugin/Analytics/AppInsightsEventTelemetry.cs:8]: Do we need this summary? The links are broken.

Comment (nemesv) [reply src/dotnet/ReSharperPlugin.SpecflowRiderPlugin/Analytics/AppInsightsEventTelemetry.cs:8]: Good catch, I will try to fix the links

File: src/dotnet/ReSharperPlugin.SpecflowRiderPlugin/Analytics/RiderInstallationStatus.cs [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+    {
+        public string InstalledVersion { get; set; }
+        
+        public DateTime? InstallDate { get; set; }
+        
+        public DateTime? LastUsedDate { get; set; }
+        
+        public int UsageDays { get; set; }
+        
+        public int UserLevel { get; set; }
```
[the remark this answers is not in the document]
Comment (nemesv) [reply src/dotnet/ReSharperPlugin.SpecflowRiderPlugin/Analytics/RiderInstallationStatus.cs:15]: These need to stay, we will need them when implementing onboarding messages.

[the remark this answers is not in the document]
Comment (alexandrafung) [reply src/dotnet/ReSharperPlugin.SpecflowRiderPlugin/Analytics/RiderInstallationStatus.cs:15]: Ok, thanks.

File: src/dotnet/ReSharperPlugin.SpecflowRiderPlugin/Analytics/AppInsightsEventTelemetry.cs (hunk 1 above)
Comment (nemesv) [reply src/dotnet/ReSharperPlugin.SpecflowRiderPlugin/Analytics/AppInsightsEventTelemetry.cs:8]: Fixed

**Pull request merged**